src/types/TypedFunctionType.ts

```diff
@@ -81,12 +86,13 @@ export class TypedFunctionType extends BaseFunctionType {
 
     isEqual(targetType: BscType, data: TypeCompatibilityData = {}) {
         if (isTypedFunctionType(targetType)) {
-            if (this.toString().toLowerCase() === targetType.toString().toLowerCase()) {
+            const checkNames = data?.allowNameEquality ?? true;
```

Comment:
Usually, a function type in the same scope is the same if the signature is the same - but that's not guaranteed when comparing callfuncs. There might be params with certain interface types that have the same name, but mean something different